### PR TITLE
vcfutils: add bcf_remove_alleles_set

### PR DESCRIPTION
**Pull request by mcshane**
* new bcf_remove_alleles_set function to the new kbitset.h api to indicated alleles
  to be removed. This avoids overflowing the old integer bitmask in bcf_remove_alleles
  when there are more than 31 alleles
* bcf_remove_alleles updated to use the new bcf_remove_alleles_set
* Fixes samtools/bcftools#319

### CODE REVIEW

File: htslib/vcfutils.h [hunk 1]
```diff
@@ -1,6 +1,6 @@
 /*  vcfutils.h -- allele-related utility functions.
 
-    Copyright (C) 2012, 2013 Genome Research Ltd.
+    Copyright (C) 2012-2015 Genome Research Ltd.
```
Comment (jmarshall) [inline htslib/vcfutils.h]: Not sure there were copyright-significant changes in 2014

File: vcfutils.c [hunk 2]
```diff
@@ -198,19 +198,32 @@ int bcf_trim_alleles(const bcf_hdr_t *header, bcf1_t *line)
...
+    int i;
+    kbitset_t *bset = kbs_init(line->n_allele);
+    for (i=1; i<line->n_allele; i++)
+        if ( rm_mask & 1<<i ) kbs_insert(bset, i);
+
+    bcf_remove_alleles_set(header, line, bset);
+    kbs_destroy(bset);
+}
+
+void bcf_remove_alleles_set(const bcf_hdr_t *header, bcf1_t *line, kbitset_t *bset)
```
Comment (jmarshall) [inline vcfutils.c]: const kbitset_t *

File: htslib/vcfutils.h [hunk 3]
```diff
@@ -26,6 +26,7 @@ DEALINGS IN THE SOFTWARE.  */
 #define HTSLIB_VCFUTILS_H
 
 #include "vcf.h"
+#include "htslib/kbitset.h"
```
Comment (jmarshall) [inline htslib/vcfutils.h]: Changing includes ⇒ changing dependencies in the makefile.  Also what I was suggesting in _Example of declaring…_ in kbitset.h was to not include the kbitset header here, but for this header to have

    struct kbitset;

here (actually, just below the `extern "C"` bit) and

    void bcf_remove_alleles_set(const bcf_hdr_t *header, bcf1_t *line, const struct kbitset_t *rm_set);
                                                                             ^^^^^^^^^^^^^^^^

below.  Only _vcfutils.c_ actually needs to `#include <htslib/kbitset.h>` and correspondingly have `$(htslib_kbitset_h)` in its objects' dependencies.

**Comment by mcshane** [comment]
Updated PR after @jmarshall comments.

**Comment by jmarshall** [comment]
:+1:

(Is it `bcf_remove_alleles_set` or `bcf_remove_allele_set`?)

**Comment by mcshane** [comment]
I'm not sure of the *set context you and James have in mind, so I'm reading it as remove alleles (plural) that have the bit set.

**Comment by jmarshall** [comment]
I was thinking of a _(mathematical) set of alleles_… which I would write as `allele_set`, but YMMV.

**Comment by mcshane** [comment]
You have a supporter in @daviesrob, so changed to `bcf_remove_allele_set`

**Pull request merged**